### PR TITLE
[tl_classifier] Add 2 models and classifier finally

### PR DESCRIPTION
**Pull request by anonymint**
* [tl_classifier] Update class to include Tensorflow and 2 models file 
* [tl_detector] replace simple `light.state` with classifier

### CODE REVIEW

File: ros/src/tl_detector/light_classification/tl_classifier.py [hunk 1]
```diff
@@ -1,9 +1,41 @@
 from styx_msgs.msg import TrafficLight
+import tensorflow as tf
+import numpy as np
+import datetime
+import rospy
 
+THRESHOLE = .5
```
Comment (soniclue) [inline ros/src/tl_detector/light_classification/tl_classifier.py]: typo: _THRESHOLD

File: ros/src/tl_detector/tl_detector.py [hunk 2]
```diff
@@ -144,17 +145,17 @@ def get_light_state(self, light):
...
         # This block will be back with Classifier
-        # if(not self.has_image):
-        #     self.prev_light_loc = None
-        #     return False
-        #
-        # cv_image = self.bridge.imgmsg_to_cv2(self.camera_image, "bgr8")
-        #
-        # #Get classification
-        # return self.light_classifier.get_classification(cv_image)
+        if(not self.has_image):
```
Comment (soniclue) [inline ros/src/tl_detector/tl_detector.py]: if not self.has_image:

File: ros/src/tl_detector/tl_detector.py [hunk 3]
```diff
@@ -47,7 +47,8 @@ def __init__(self):
         self.upcoming_red_light_pub = rospy.Publisher('/traffic_waypoint', Int32, queue_size=1)
 
         self.bridge = CvBridge()
-        self.light_classifier = TLClassifier()
+        self.is_site = self.config["is_site"]
```
Comment (soniclue) [inline ros/src/tl_detector/tl_detector.py]: is self.is_site used anywhere else?
if not, we don't need to store it as a class member variable. just pass self.config['is_site'] to TLClassifier

File: ros/src/tl_detector/light_classification/tl_classifier.py [hunk 4]
```diff
@@ -15,5 +47,34 @@ def get_classification(self, image):
...
+        #  2: {'id': 2, 'name': 'Red'},
+        #  3: {'id': 3, 'name': 'Yellow'},
+        #  4: {'id': 4, 'name': 'off'}
+        # }
+        if scores[0] >= THRESHOLE:
+            if classes[0] == 1:
+                return TrafficLight.GREEN
+            elif classes[0] == 2:
+                return TrafficLight.RED
+            elif classes[1] == 3:
```
Comment (soniclue) [inline ros/src/tl_detector/light_classification/tl_classifier.py]: classes[0] or [1]?

File: ros/src/tl_detector/light_classification/tl_classifier.py [hunk 5]
```diff
@@ -15,5 +47,34 @@ def get_classification(self, image):
             int: ID of traffic light color (specified in styx_msgs/TrafficLight)
 
         """
-        #TODO implement light color prediction
+        start_clock = datetime.datetime.now()
+        with self.graph.as_default():
+            image_expanded = np.expand_dims(image, axis=0)
+            (boxes, scores, classes, num) = self.sess.run(
```
Comment (soniclue) [inline ros/src/tl_detector/light_classification/tl_classifier.py]: no need to have () around the left hand values.

**Review by soniclue — APPROVED** [review]
I have not tried it. But feel free merge to master after addressing the comments.

File: ros/src/tl_detector/light_classification/tl_classifier.py (hunk 4 above)
Comment (anonymint) [reply ros/src/tl_detector/light_classification/tl_classifier.py]: Thanks! that's why Yellow isn't working properly to me

**Comment by anonymint** [comment]
Addressed some of the comment from @yangchunluo  and I'll go ahead and merge

**Pull request merged**